instagram-app/src/components/navigation/SearchBar.js

```diff
@@ -5,20 +5,22 @@ import './SearchBar.css';
     
     return (
           <div className="search-bar">
-              <i class="fab fa-instagram"></i>
+              <i className="fab fa-instagram"></i>
               <h6>Instagram</h6>
-              <div>
 
+              <div>
+            
               <input onKeyDown={props.searchPosts}
                   type="text"
                   name="input"
                   id="new-search"
```

Comment:
`onChange` would be more suitable here to ensure that changes in the input text will immediately take effect upon *any* sort of change in the search bar (not just the keyboard, but any source of change in input). This also ensures that when your `App.js` component checks for the length of your `searchPosts` array, it will only do so after the changes to the search bar input have been properly registered.

Using the `onChange` event attribute also avoids a glitch in your original implementation for Day 2 wherein the search filter will be late by one key press in taking effect, and this also prevents changes in search results from being erroneously triggered by a key press that does not involving typing in an actual alphanumeric character or special symbol.